app/gui/src/controller/graph/executed.rs

```diff
@@ -175,6 +176,11 @@ impl Handle {
         self.execution_ctx.modify_visualization(id, Some(code), Some(module)).await
     }
 
+    /// See [`model::ExecutionContext::component_groups`].
+    pub fn component_groups(&self) -> Rc<Vec<ComponentGroup>> {
+        self.execution_ctx.component_groups()
+    }
```

Comment:
Why we don't either:
1. Users accessing `execution_ctx`?
2. Deref to `execution_ctx`?